salidl/Dockerfile

```diff
@@ -0,0 +1,67 @@
+ARG OSPL_VERSION=6.9.0
+FROM lsstdm/salrun:${OSPL_VERSION} AS salidl-build
+
+RUN yum -y install java-1.7.0-openjdk-headless \
+    && mkdir -p /etc/alternatives/java_sdk_openjdk/jre \
+    && ln -s /etc/alternatives/jre_openjdk/bin /etc/alternatives/java_sdk_openjdk/bin \
+    && ln -s /etc/alternatives/jre_openjdk/lib /etc/alternatives/java_sdk_openjdk/jre/lib
+
+ENV PYTHON_BUILD_VERSION=3.6m \
+    PYTHON_BUILD_LOCATION=/usr \
+    LSST_SDK_INSTALL=/tmp/build/ts_sal \
+    SAL_WORK_DIR=/tmp/build/ts_sal/test
+
+ARG LSSTTS_IDL_VERSION
+
+WORKDIR /tmp/build
+RUN git clone https://github.com/lsst-ts/ts_idl.git \
+    && cd ts_idl \
+    && git checkout v$LSSTTS_IDL_VERSION
+
+ARG LSSTTS_SAL_VERSION
+
+RUN git clone https://github.com/lsst-ts/ts_sal.git \
+    && cd ts_sal \
+    && git checkout v$LSSTTS_SAL_VERSION
+
+ARG LSSTTS_XML_VERSION
+
+RUN git clone https://github.com/lsst-ts/ts_xml.git \
+    && cd ts_xml \
+    && git checkout v$LSSTTS_XML_VERSION \
+    && ln -s /tmp/build/ts_xml/sal_interfaces/SALGenerics.xml $SAL_WORK_DIR \
+    && ln -s /tmp/build/ts_xml/sal_interfaces/SALSubsystems.xml $SAL_WORK_DIR \
+    && ln -s /tmp/build/ts_xml/sal_interfaces/*/*xml $SAL_WORK_DIR
+
+ARG CCS_IDL_LIST
+
+RUN cd $SAL_WORK_DIR \
+    && source /opt/lsst/setup_SAL.env \
+    && source /tmp/build/ts_sal/setup.env \
+    # && for csc in `cd /tmp/build/ts_xml/sal_interfaces; ls | grep -v xml`; do \
+    # Running salgenerator for all CSCs takes ~4hrs, so we'll only
+    # build the ones we strickly need
+    && for csc in $CCS_IDL_LIST; \
+       do \
+           salgenerator $csc validate > /dev/null && echo $csc validate done || exit 1; \
+           salgenerator $csc html > /dev/null && echo $csc html done || exit 1; \
+           salgenerator $csc sal cpp > /dev/null && echo $csc sal cpp done || exit 1; \
+       done
+
+# These are the ones generated by the 'validate' command
+RUN mv /tmp/build/ts_sal/test/idl-templates/validated/*.idl /tmp/build/ts_idl/idl/
+# These are the ones needed by salobj and generated after the cpp build
+RUN mv /tmp/build/ts_sal/test/idl-templates/validated/sal/sal_revCoded_*.idl /tmp/build/ts_idl/idl/
+
+FROM lsstdm/salrun:${OSPL_VERSION}
```

Comment:
It seems strange to have the faster-changing IDL be the basis for the slower-changing salobj container, rather than the other way around.

Reply:
That's right, the `ts_idl` module changes faster than `saobj` as it depends on `ts_xml`. Unfortunately `ts_idl` is a pre-req of `salobj` and therefore having an `salobj` docker  image without `ts_idl` in it make such `salobj` docker imahe unusable (basically broken). I considered that fact important enought to change the order to current one. If you don't think that's the case, we can revert it to the previous order.

Reply:
Adding the IDL files on top of an underlying `salobj` container without having to rebuild the latter each time makes some sense to me.  The point wasn't to have `salobj` usable as is; if you need a usable-out-of-the-box container, `salidl` would be the one.  The point was to factor out the slowly-changing layers so they only needed to be rebuilt occasionally.

Now, if building `salidl` takes a long time because the complex `salgenerator` subcommands need to be used, then there isn't much savings by doing this.  But once the IDL files are RPM-packaged, it should be much easier to rebuild `salidl`.

Reply:
Yes, once the IDL files are bundled with the rpms this should be easier and can be shuffled. At the moment `salidl` takes a long time to build.